clang/include/clang/Analysis/Analyses/LifetimeSafety/Origins.h

```diff
@@ -16,7 +16,16 @@
 
 #include "clang/AST/Decl.h"
 #include "clang/AST/Expr.h"
+#include "clang/AST/RecursiveASTVisitor.h"
+#include "clang/AST/TypeBase.h"
 #include "clang/Analysis/Analyses/LifetimeSafety/Utils.h"
+#include "llvm/ADT/StringMap.h"
+#include "llvm/Support/raw_ostream.h"
+
+namespace clang::lifetimes {
+
+struct LifetimeSafetyStats;
```

Comment:
Ah sorry for missing this in my original suggestion. I think you have added this because it creates a cyclic header includes if we include LifetimeSafety.h here

Maybe have a separate header for Stats "clang/include/clang/Analysis/Analyses/LifetimeSafety/Stats.h" `printStats`​ can live in clang/lib/Analysis/LifetimeSafety/Stats.cpp
`collectLifetimeStats`​ would need to still live in LifetimeSafety.cpp